apps/server/src/modules/files/repo/sync-files.repo.ts

```diff
@@ -0,0 +1,121 @@
+/* istanbul ignore file */
+
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId, FileRecord, FileRecordParentType, Task } from '@shared/domain';
+import { SyncFileItemMapper } from '../mapper';
+import { SyncFileItem, SyncFileItemData } from '../types';
+
+// This repo is used for syncing the new filerecords collection with the old files collection.
+// It can be removed after transitioning file-handling to the new files-storage-microservice is completed.
+@Injectable()
+export class SyncFilesRepo {
+	constructor(protected readonly _em: EntityManager) {}
+
+	async findTaskFilesToSync(batchSize = 50): Promise<SyncFileItem[]> {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const itemDataList: SyncFileItemData[] = await this._em.aggregate(Task, [
```

Comment:
Split the aggegration it self to a query builder in our code called Scope. Then we can use it similar for 
await this._em.aggregate(Task, scope.query);
await this._em.aggregate(Submission, scope.query);